backends/fs/android/android-posix-fs.cpp

```diff
@@ -0,0 +1,61 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#if defined(__ANDROID__)
+
+#include "backends/fs/android/android-fs-factory.h"
+#include "backends/fs/android/android-posix-fs.h"
+#include "backends/fs/android/android-saf-fs.h"
+
+AbstractFSNode *AndroidPOSIXFilesystemNode::makeNode() const {
+	return new AndroidPOSIXFilesystemNode(_config);
+}
+
+AbstractFSNode *AndroidPOSIXFilesystemNode::makeNode(const Common::String &path) const {
+
+	if (path.equals(AddSAFFakeNode::SAF_ADD_FAKE_PATH)) {
+		return new AddSAFFakeNode();
+	}
+
+	// If SAF works, it was a SAF URL
+	AbstractFSNode *node = AndroidSAFFilesystemNode::makeFromPath(path);
+	if (node) {
+		return node;
+	}
+
+	return new AndroidPOSIXFilesystemNode(path, _config);
+}
+
+AndroidPOSIXFilesystemNode::AndroidPOSIXFilesystemNode(const Common::String &path,
+        const Config &config) :
+	DrivePOSIXFilesystemNode(path, config) {
+	if (((uintptr_t)&path >= (uintptr_t)config.drives.begin()) &&
```

Comment:
Could you add a comment here explaining what this does?

Reply:
Sure, done.